pkg/controller/controller.go

```diff
@@ -94,12 +98,13 @@ type ReQueue interface {
 }
 
 // NewQueueRetryManager safely creates a new QueueRetryManager.
-func NewQueueRetryManager(queue ReQueue, keyFn kcache.KeyFunc, retryFn RetryFunc) *QueueRetryManager {
+func NewQueueRetryManager(queue ReQueue, keyFn kcache.KeyFunc, retryFn RetryFunc, rate float32, burst int) *QueueRetryManager {
 	return &QueueRetryManager{
 		queue:     queue,
 		keyFunc:   keyFn,
 		retryFunc: retryFn,
 		retries:   make(map[string]int),
+		limiter:   kutil.NewTokenBucketRateLimiter(rate, burst),
```

Comment:
> On Apr 3, 2015, at 10:18 PM, Ben Parees <notifications@github.com> wrote:
> 
> In pkg/controller/controller.go:
> 
> >  	return &QueueRetryManager{
> >  		queue:     queue,
> >  		keyFunc:   keyFn,
> >  		retryFunc: retryFn,
> >  		retries:   make(map[string]int),
> > +		limiter:   kutil.NewTokenBucketRateLimiter(rate, burst),
> it also means any consumers of this now have to import, understand, and instantiate, another type instead of two primitives. i'm not convinced that's an improvement from a usability+readability perspective.
> 
Eh... Two arguments that you have to explain are not any worse to me than a copy and paste constructor (which is going to be pretty name clear).  Taking the interface is straight up better - it means if you want a complex rate limiter you don't have to change the class.  Also, Factory is kind of a "configure a bunch of things" including other lower level bits.  I don't like encapsulating function that isn't specific to the class (the "how" of limiting is orthogonal to the need to limit).
> I like that the RateLimiter interface is currently encapsulated so users of QueueRetryManager don't need to be aware of it, the potential benefits w/ respect to test code (that you mention below) not withstanding.
> 
> —
> Reply to this email directly or view it on GitHub.
>